### PR TITLE
tcp_proxy: add README with deprecated v1 example

### PR DESCRIPTION
**Pull request by zuercher**
Configuration of the TCP proxy's deprecated source ip and port
feature comes up periodically and since it was non-trivial for
me to figure out how to even configure it, I thought I'd add
a README with an example.

*Risk Level*: low, docs only
*Testing*: manually tested example
*Docs Changes*: added example for TCP proxy deprecated_v1
*Release Notes*: n/a

Signed-off-by: Stephan Zuercher \<zuercher@gmail.com\>

### CODE REVIEW

File: api/envoy/config/filter/network/tcp_proxy/v2/README.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
+Protocol buffer definitions for the TCP Proxy.
```
Comment (mattklein123) [inline api/envoy/config/filter/network/tcp_proxy/v2/README.md]: Is there any reason to not just do this inline in the docs in the filter configuration section? We have other samples in there and an MD seems a bit out of place?

File: api/envoy/config/filter/network/tcp_proxy/v2/tcp_proxy.proto [hunk 2]
```diff
@@ -136,13 +134,42 @@ message TcpProxy {
...
-  //   Using this field will lead to `problems loading the configuration
-  //   <https://github.com/envoyproxy/envoy/issues/2441>`_. If you want to configure the filter
-  //   using v1 config structure, please make this field a boolean with value ``true`` and configure
-  //   via the opaque ``value`` field like is suggested in
-  //   :api:`envoy/config/filter/network/tcp_proxy/v2/README.md`.
+  //   Using the deprecated V1 configuration excludes the use of any V2 configuration options. Only
+  //   the V1 configuration is used. All available fields are shown in the example, although the
+  //   access log configuration is omitted for simplicity. The access log configuration uses the
+  //   `deprecated V1 access log configuration
+  //   <https://github.com/envoyproxy/envoy/blob/master/source/common/json/config_schemas.cc>`_.
```
Comment (venilnoronha) [inline api/envoy/config/filter/network/tcp_proxy/v2/tcp_proxy.proto]: :repo:\`source/common/json/config_schemas.cc\`

**Comment by zuercher** [comment]
@mattklein123 I don't think [TCPProxy.DeprecatedV1[(https://www.envoyproxy.io/docs/envoy/latest/api-v2/config/filter/network/tcp_proxy/v2/tcp_proxy.proto.html?highlight=tcpproxy#config-filter-network-tcp-proxy-v2-tcpproxy-deprecatedv1) actually gets used and it incorrectly implies that you can specify V1 configs using CIDRRange structs when, in fact, you have to provide a list of strings (e.g. "10.1.0.0/16" vs `{ "address_prefix":"10.1.0.0", "prefix_len": 16 }`. Does it make sense to change it as well?

**Pull request merged**